dist/node-jsonapi-datastore.js

```diff
@@ -236,9 +254,30 @@
               if (rel.data === null) {
                 model[key] = null;
               } else if (rel.data.constructor === Array) {
-                model[key] = rel.data.map(findOrInit);
+                model[key] = rel.data.map(findOrInit).map(function(record) {
+                  var found = record._dependents.filter(function(dependent) {
+                    return dependent.id === model.id && dependent.type === model._type;
+                  });
+                  if (found.length === 0) {
+                    record._dependents.push({
+                      id: model.id,
+                      type: model._type,
+                      relation: key
+                    });
+                  }
+                });
```

Comment:
Nice update, I kinda overlooked that case when I reviewed the previous version. This is getting a bit messy though. How about we define `_addDependence(type, id, key)` on the Model? It would allow for a simple
```javascript
model[key] = rel.data.map(findOrInit);
model[key].forEach(function(record) { record._addDependence(model._type, model.id, key); });
```